### DISCUSSION TITLE
Welcome to Telegram Vapor Bot Discussions!

### DISCUSSION THREAD

**Discussion by nerzh** [nerzh/telegram-vapor-bot#10] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Holejoek** [comment]
@nerzh  do u try to launch bot with guides? Its so hard for beginners. Readme in package and readme on github are different.
 Due to discrepancies in the readme files and guides I had to waste a lot of time. As a result, having reached the error that is displayed on the screenshot, for today I gave up in this struggle. 
<img width="1196" alt="Снимок экрана 2023-04-26 в 00 59 12" src="https://user-images.githubusercontent.com/91912185/234390284-2a3e84ef-6a51-4a49-8daa-687628635448.png">
The screenshot above shows the implementation of "main.swift" from the readme file from 25/04/23


BUT, I want to thank the author for the opportunity to open the doors to the world of interaction with telegram bots via SWIFT 😇

**Reply by Butanediol** [reply]
Do you have any experience with Vapor? The readme is confusing for people who are not familiar with Vapor.

If not, I strongly recommend reading Vapor docs first to get some basic ideas of how to initialise a Vapor project.

This library is meant to use with Vapor. I think it's actually quite straightforward compared to other Telegram bot libs because Vapor handles a lot of things.

And if you have any other questions, feel free to ask. I really like this library as well!

**Reply by Holejoek** [reply]
In addition, I was unlucky enough to get acquainted with Vapor during its update 

<img width="522" alt="Снимок экрана 2023-04-26 в 12 54 52" src="https://user-images.githubusercontent.com/91912185/234508217-fb3ba9d2-798a-4e41-8a92-ac632e3a44d4.png">

https://discord.com/channels/431917998102675485/1100685582541279314

**Reply by Butanediol** [reply]
Oops, this is unexpected. I now have to read Vapor manual too.

**Comment by nerzh** [comment]
@Holejoek Hello, have you tried running a ready-made bot template ? 
https://github.com/nerzh/telegram-vapor-bot/tree/master/Telegram-bot-example

**Reply by Holejoek** [reply]
Yes it works) But according to the guide for beginners and another stuff here, it doesn't work. Perhaps I really need to deal with Vapor as @Butanediol  advised me. 
Just in case, my experience in development is 1.5 years. I hope no one here cringed, and my question was useful 🙃

**Comment by Holejoek** [comment]


**Reply by Butanediol** [reply]
There shouldn’t be any problems to do so. Are you sure you registered those routes properly? 🤔

**Comment by Holejoek** [comment]
I have a problem again, please help me) I've been struggling with this for about a week now. The server starts up on Ubuntu 22.04 using Docker and everything works fine, but after a while, after about half a day, without stopping the running server, the telegram bot simply stops responding - I set the logger level to .debug and that's what I got:
```
[ DEBUG ] SELECT "role_table"."id" AS "role_table_id", "role_table"."userId" AS "role_table_userId", "role_table"."role" AS "role_table_role", "role_table"."name" AS "role_table_name" FROM "role_table" WHERE "role_table"."userId" = $1 LIMIT 1 [[389266933]] [database-id: psql] (PostgresKit/PostgresDatabase+SQL.swift:44)
[ DEBUG ] query aggregate(count(order[id])) order filters=[order[userId] = 389266933, order[state] != canceled, order[state] != created] [database-id: psql] (FluentKit/QueryBuilder.swift:293)
[ DEBUG ] SELECT COUNT("order"."id") AS "aggregate" FROM "order" WHERE "order"."userId" = $1 AND "order"."state" <> 'canceled' AND "order"."state" <> 'created' [[389266933]] [database-id: psql] (PostgresKit/PostgresDatabase+SQL.swift:44)
[ DEBUG ] query aggregate(count(act_counter[id])) act_counter filters=[act_counter[userId] = 389266933, act_counter[code] = 0, act_counter[time] >= 1694884801561] [database-id: psql] (FluentKit/QueryBuilder.swift:293)
[ DEBUG ] SELECT COUNT("act_counter"."id") AS "aggregate" FROM "act_counter" WHERE "act_counter"."userId" = $1 AND "act_counter"."code" = $2 AND "act_counter"."time" >= $3 [[389266933, 0, 1694884801561]] [database-id: psql] (PostgresKit/PostgresDatabase+SQL.swift:44)
[ DEBUG ] query create act_counter input=[[time: 1695489601567, code: 0, id: 06935C35-D575-4D36-B572-4ED7C7CBC48B, userId: 389266933]] [database-id: psql] (FluentKit/QueryBuilder.swift:293)
[ DEBUG ] INSERT INTO "act_counter" ("id", "userId", "code", "time") VALUES ($1, $2, $3, $4) RETURNING "id" [[06935C35-D575-4D36-B572-4ED7C7CBC48B, 389266933, 0, 1695489601567]] [database-id: psql] (PostgresKit/PostgresDatabase+SQL.swift:44)
[ DEBUG ] connection go away received [ahc-connection-id: 0, ahc-http-version: http/2, ahc-pool-key: https://api.telegram.org:443 TLS-hash: 4896907730711376828 ] (AsyncHTTPClient/HTTPConnectionPool.swift:517)
[ DEBUG ] connection closed [ahc-connection-id: 0, ahc-http-version: http/2, ahc-pool-key: https://api.telegram.org:443 TLS-hash: 4896907730711376828 ] (AsyncHTTPClient/HTTPConnectionPool.swift:527)
[ DEBUG ] Request was queued (waiting for a connection to become available) [ahc-request-id: 1001] (AsyncHTTPClient/RequestBag.swift:89)
[ DEBUG ] connection go away received [ahc-connection-id: 1, ahc-http-version: http/2, ahc-pool-key: https://api.telegram.org:443 TLS-hash: 4896907730711376828 ] (AsyncHTTPClient/HTTPConnectionPool.swift:517)
[ DEBUG ] connection closed [ahc-connection-id: 1, ahc-http-version: http/2, ahc-pool-key: https://api.telegram.org:443 TLS-hash: 4896907730711376828 ] (AsyncHTTPClient/HTTPConnectionPool.swift:527)
[ DEBUG ] Request was queued (waiting for a connection to become available) [ahc-request-id: 2001] (AsyncHTTPClient/RequestBag.swift:89)
[ ERROR ] 
>>>Type: server
>>>Description: Response marked as `not Ok`, it seems something wrong with request
Code: 502
Bad Gateway
>>>Reason: 
 (TelegramVaporBot/TGClient.swift:145)
[ DEBUG ] connection closed [ahc-connection-id: 2, ahc-http-version: http/2, ahc-pool-key: https://api.telegram.org:443 TLS-hash: 4896907730711376828 ] (AsyncHTTPClient/HTTPConnectionPool.swift:527)
```

**Reply by Holejoek** [reply]
No one but me communicates with the telegram bot, so the load is minimal and not frequent

**Reply by nerzh** [reply]
where is your server running? what is your provider?

**Reply by Holejoek** [reply]
VPS 2Gb RAM,  2CPU - https://www.reg.ru/cloud/vps/

**Reply by nerzh** [reply]
hmm, the logger writes to you that this is a network server error 502 bad gateway. Are you using longpolling or webhooks?

**Reply by Holejoek** [reply]
longpolling
